### PR TITLE
iterate over retry_delays

### LINKED ISSUE #171: lsf_bpeek makes strong assumptions on iterator state of retry_delays
[This line](https://github.com/JuliaParallel/ClusterManagers.jl/blob/70f443426be99269251a151451acb65c6275aa11/src/lsf.jl#L48) seems to make some pretty strong assumptions on the retry_delays. 

This is somewhat compounded by the fact that the error message does not have a stacktrace, so it is not easy for a user to figure out what is wrong:

```julia
julia> addprocs_lsf(10; retry_delays = Base.Iterators.cycle(5))
Error launching workers
MethodError(getindex, (nothing, 1), 0x00000000000073d6)
Int64[]
```

Is the intention of the line to just catch when there are no more backoffs to try? If so, why not just check for when iterator returns nothing or rewrite loop to use iterator?

### PR DESCRIPTION
**Pull request by DrChainsaw**
Fixes #171 

I have run some manual testing to add procs as well as testing with with a non-public package which depends on ClusterManagers.

@tanmaykm I'm not 100% I understood all aspects of the code so I'd be happy if you can take a look at this. I suppose there are alot of edge cases which might not be covered by my testing.

### CODE REVIEW

File: src/lsf.jl [hunk 1]
```diff
@@ -63,6 +58,11 @@ function lsf_bpeek(manager::LSFManager, jobid, iarray)
         end
     end
 
+    if !worker_started
+        close(stream)
+        throw(LSFException(bytestr))
```
Comment (tanmaykm) [inline src/lsf.jl:63]: Looks like `bytestr` is defined inside the `for` loop body, so this may throw `UndefVarError`?

**Comment by tanmaykm** [comment]
Yes, using the iterator seems a lot better. Thanks!

File: src/lsf.jl (hunk 1 above)
Comment (DrChainsaw) [reply src/lsf.jl:67]: Ooops, good catch! Should be fixed now.

**Comment by DrChainsaw** [comment]
Thanks!

I rewrote the logic so it tries to parse the host and port before going into the retry loop. Besides resolving the bytestr scope issue it should also do the expected thing if the user supplies an empty iterator for the retry_delays.

**Comment by DrChainsaw** [comment]
Bump.

**Pull request merged**